hooks/checkout

```diff
@@ -38,7 +38,11 @@ checkout() {
   git fetch -v origin HEAD
   git config remote.origin.fetch
   git fetch -v --prune origin "${BUILDKITE_BRANCH}"
-  git checkout -f "${BUILDKITE_COMMIT}"
+  if [[ "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
+    git checkout -f FETCH_HEAD
```

Comment:
Will `FETCH_HEAD` here point to the head of the `BUILDKITE_BRANCH` we fetched above, even if we didn't check it out?

Reply:
yes

Reply:
The names of refs that are fetched, together with the object names they point at, are written to .git/FETCH_HEAD